### PR TITLE
Introduce new method in circuit breaker factory

### PR DESCRIPTION
**Pull request by Ferioney**
Related to [Spring Cloud OpenFeign Issue 514](https://github.com/spring-cloud/spring-cloud-openfeign/issues/514)

### CODE REVIEW

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/circuitbreaker/CircuitBreakerFactory.java [hunk 1]
```diff
@@ -20,10 +20,13 @@
...
  *
  * @author Ryan Baxter
+ * @author Andrii Bohutskyi
  */
 public abstract class CircuitBreakerFactory<CONF, CONFB extends ConfigBuilder<CONF>>
 		extends AbstractCircuitBreakerFactory<CONF, CONFB> {
 
 	public abstract CircuitBreaker create(String id);
 
+	public abstract CircuitBreaker create(String circuitBreakerName, String bulkheadName);
```
[the remark this answers is not in the document]
Comment (Ferioney) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/circuitbreaker/CircuitBreakerFactory.java]: Changed to `groupName`

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/circuitbreaker/CircuitBreakerFactory.java [hunk 2]
```diff
@@ -20,10 +20,13 @@
...
  *
  * @author Ryan Baxter
+ * @author Andrii Bohutskyi
  */
 public abstract class CircuitBreakerFactory<CONF, CONFB extends ConfigBuilder<CONF>>
 		extends AbstractCircuitBreakerFactory<CONF, CONFB> {
 
 	public abstract CircuitBreaker create(String id);
 
+	public abstract CircuitBreaker create(String id, String groupName);
```
Comment (spencergibb) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/client/circuitbreaker/CircuitBreakerFactory.java:30]: Also, it probably can not be abstract (as existing implementations will break). The default implementation should just call the other method.

**Review by spencergibb — CHANGES_REQUESTED** [review]

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/circuitbreaker/CircuitBreakerFactory.java (hunk 2 above)
Comment (Ferioney) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/circuitbreaker/CircuitBreakerFactory.java]: Good point. Changed

**Review by spencergibb — APPROVED** [review]

**Comment by Ferioney** [comment]
\+ @ryanjbaxter

**Comment by ryanjbaxter** [comment]
@Ferioney if you want this in Hoxton, you would need to submit it against the 2.2.x branch

**Comment by Ferioney** [comment]
@ryanjbaxter PR to 2.2.x https://github.com/spring-cloud/spring-cloud-commons/pull/941
Would be nice to have those changes in 2020 and Hoxton

**Pull request closed without merging**